### ISSUE TITLE
Implement t-SNE VISUALISATIOn during training and x3d

### ISSUE THREAD

**Issue by robertavram-md** [HeartWise-AI/DeepCORO_CLIP#50]
The dimensionality reduction (t-SNE, UMAP) to visualize the video and text embeddings from the validation set at different checkpoints (before peak, at peak, after peak). This can give insights into how the embedding space structure changes during overfitting.
